infra/staging/main.tf

```diff
@@ -151,6 +151,66 @@ resource "google_storage_bucket_iam_member" "frontend_bucket_viewer" {
   role   = "roles/storage.objectViewer"
   member = "allUsers"
 }
+#######################################
+## Load balancer for frontend bucket ##
+#######################################
+
+// Following the guide here: https://cloud.google.com/cdn/docs/setting-up-cdn-with-bucket
+
+// Enable Compute Engine API to setup an IP address
+resource "google_project_service" "compute_engine_api" {
```

Comment:
I don't know why exactly this is needed, but I'm willing to accept that it's needed to setup an IP address _just because_. GCP has a bunch of weird legacy problems with APIs and permissions, so I wouldn't be surprised if this is one of those.